web-app/src/main/webapp/script.js

```diff
@@ -19,25 +19,51 @@ function initMap() {
   });
   const infoWindow = new google.maps.InfoWindow();
   displayUserLocation(map, infoWindow);
-  fetchMarkers();
+  fetchMarkers(map);
 }
 
-async function fetchMarkers() {
+async function fetchMarkers(map) {
   const response = await fetch('/report');
   const markers = await response.json();
+  let openWindow = { window: null };
+
   markers.forEach((marker) => {
-    console.log(marker);
-    createMarkerForDisplay(marker);
+    createMarkerForDisplay(map, marker, openWindow);
   });
+
+  map.addListener('click', () => {
+    if (openWindow.window) {
+      openWindow.window.close();
+      openWindow.window = null;
+    }
+  })
 }
 
-function createMarkerForDisplay(data) {
+function createMarkerForDisplay(map, data, openWindow) {
```

Comment:
nit: `data` is getting a bit complicated now, maybe it's time to add a comment describing what fields it has.